dalesbred/src/main/java/org/dalesbred/dialect/Dialect.java

```diff
@@ -159,4 +163,41 @@ public void registerTypeConversions(@NotNull TypeConversionRegistry typeConversi
     public void bindArgument(@NotNull PreparedStatement ps, int index, @Nullable Object value) throws SQLException {
         ArgumentBinder.bindArgument(ps, index, value);
     }
+
+    public Map<String, Class<?>> getResultSetMetaDataTypeOverrides() {
+        return resultSetMetaDataTypeOverrides;
+    }
+
+    // Some drivers (e.g. MySQL and H2) return version strings that do not look like typical SemVer strings and hence will not work with this
+    protected static class Version implements Comparable<Version> {
+
+        private final int[] numbers;
+
+        private Version(@NotNull String version) {
+            final String split[] = version.split("\\-")[0].split("\\.");
+            numbers = new int[split.length];
+            for (int i = 0; i < split.length; i++)
+                numbers[i] = Integer.parseInt(split[i]);
+        }
+
+        protected static Version withNumber(@NotNull String version) {
+            return new Version(version);
+        }
+
+        @Override
+        public int compareTo(@NotNull Version another) {
+            final int maxLength = Math.max(numbers.length, another.numbers.length);
+            for (int i = 0; i < maxLength; i++) {
+                final int left = i < numbers.length ? numbers[i] : 0;
+                final int right = i < another.numbers.length ? another.numbers[i] : 0;
+                if (left != right)
+                    return left < right ? -1 : 1;
+            }
+            return 0;
+        }
+
+        protected boolean isGreaterThanOrEqualTo(Version another) {
+            return this.compareTo(another) >= 0;
+        }
```

Comment:
I think this is superfluous. A Java-programmer should see `a.compareTo(b) >= 0` and see `a >= b` in their mind. Of course it's ugly and it's nice that Kotlin does that automatically, but still I think the compareTo-idiom is so common that there's no need to add method just to hide that.